corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -279,26 +278,18 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
         },
 
         changeQueryField: function (e) {
-            if (this.model.get('input') === 'date') {
-                // Skip because dates get handled by changeDateQueryField
-                return;
-            } else if (this.model.get('input') === 'select1' || this.model.get('input') === 'select') {
-                this.model.set('value', $(e.currentTarget).val());
-            } else if (this.model.get('input') === 'address') {
+            if (this.model.get('input') === 'address') {
                 // geocoderItemCallback sets the value on the model
+            } else if (this.model.get('input') === 'checkbox') {
+                let value = this.model.get('value');
+                this.model.set('value', value !== 'checked' ? 'checked' : '');
```

Comment:
I'm curious, if the type is a checkbox, instead of setting a string value, can you set it directly to checked here? `$(e.currentTarget).checked = true;`?